pkg/controller/storagecluster/storagecluster.go

```diff
@@ -774,6 +774,12 @@ func (c *Controller) manage(
 		}
 	}
 
+	if cluster.Status.Phase == "" && cluster.Spec.CloudStorage.MaxStorageNodesPerZone == nil {
+		// Let's do this only when it's a fresh install of px
+		cluster.Spec.CloudStorage.MaxStorageNodesPerZone = new(uint32)
+		*cluster.Spec.CloudStorage.MaxStorageNodesPerZone = getDefaultMaxStorageNodesPerZone(zoneMap)
+	}
+
```

Comment:
This will pass the updated value to the portworx driver and create the correct px pod, but I don't think this will update the value in the storagecluster itself. 
We need to update the storagecluster object in k8s. I am thinking that may be we can move the entire zone calculation code to the SetDefaults.. method.